config/routes.rb

```diff
@@ -19,6 +19,12 @@
                           as: :true_destroy_user_session
   end
 
+  #UserProfilesController
+  controller :user_profiles do
+    get 'user_profiles/:username' => 'user_profiles#show' , constraints: { username: /.*/ }
```

Comment:
We want to keep the same route `/users/:username`. Otherwise, it will break links and cause confusion.